src/components/Router/index.js

```diff
@@ -446,35 +446,17 @@ class HistoryRouter extends Component {
   }
 
   formattedError = (response, status) => {
-    if (typeof response === 'string') {
-      // TODO: this if statement should be deleted once all APIs start using the same signature for responses
-      // Currently detect_documents returns just a string. Examples:
-      // `Could not decode token: hello`
-      // `Token has expired.`
-      // Telephony returns the a JSON response
-      // {“unauthorized”:”Could not decode token: hello"}
-      // {"unauthorized":"Token has expired."}
-      // Tickets in backlog to update all APIs to use signature similar to main Onfido API
-      let message
-      try {
-        const jsonRes = JSON.parse(response)
-        message = jsonRes.unauthorized || jsonRes.error || response
-      } catch {
-        // response is just a string so we will return it as the message
-        message = response
-      }
-      const type = message.includes('expired') ? 'expired_token' : 'exception'
-      return { type, message }
-    }
-    const apiError = response.error || {}
+    const errorResponse = response.error || response || {}
+    // TODO: remove once find_document_in_image back-end `/validate_document` returns error response with same signature
+    const isExpiredTokenErrorMessage =
+      typeof response === 'string' && response.includes('expired')
     const isExpiredTokenError =
-      status === 401 && apiError.type === 'expired_token'
+      status === 401 &&
+      (isExpiredTokenErrorMessage || errorResponse.type === 'expired_token')
```

Comment:
It might be worth extracting the hardcoded string and the magic number here as constants for readability.

Reply:
Thanks @burakeregar, it was a mistake to use `master` as the base branch so this change should be out of scope here.